CMakeLists.txt

```diff
@@ -195,6 +202,9 @@ option(
   OFF)
 option(WITH_EXAMPLES "Whether to build examples" ON)
 
+# This requires CURL, OFF by default.
+option(WITH_EXAMPLES_HTTP "Whether to build http client/server examples" OFF)
+
```

Comment:
If `WITH_EXAMPLES` is off, `WITH_EXAMPLES_HTTP` will not be entertained even if CURL is installed ?

Reply:
Thanks.

Implemented a check to make sure WITH_EXAMPLE=ON when WITH_EXAMPLES_HTTP=ON.